README.md

```diff
@@ -5,40 +5,44 @@ Example Jupyter notebook source files for the BioData Catalyst consortium. These
 Jupyter notebook `.ipynb` files are generated from source files with the
 [callysto](https://github.com/xbrianh/callysto) library. 
 
-## Source File Schema
-The first line of each notebook source file contains the publish directive, a string with the following format:
-```
-# publish to "{my workspace name}" "{my notebook name}"
-```
-where `{my workspace name}` should be replaced with the name of the workspace, and `{my notebook name}` should be
-replaced with the name of the notebook.
+## Notebook Source Schema
+A notebook is defined by creating a directory under ${REPO_ROOT}/noteboks containing
+  - `main.py`, the source script for the notebook in [callysto](https://github.com/xbrianh/callysto) format.
+  - `requirements.txt`, pip-installable requirements needed for notebook execution.
+  - `publish.txt`, a list of destination Google Storage URLs where the notebook will be published.
 
-Notebook source files are executable Python scripts. The contents of cells are denoted with the callysto Python
-context managers:
+### The Callysto Format
+Notebook source files, `main.py`, are executable Python scripts. Contents of cells are denoted with the callysto
+Python context managers:
 ```
 with callysto.Cell("{mode}"):
 	...
 ```
 where `{mode}` is `python` or `markdown`. Python statements outside of callysto contexts will not be rendered into
-cells. Test code, or code for preparing mock environments, should not be placed in callysto contexts.
+cells. Test code or mock environments should not be placed outside callysto contexts.
 
 ## Publishing Notebooks
 Notebooks are published with make commands, e.g.
 ```
-make notebooks/prepare_igv_viewer_input.py
+make publish/byod
 ```
 
-This command passes the source script through the [flake8](https://flake8.pycqa.org/en/latest/) linter and
-[mypy](https://mypy.readthedocs.io/en/stable/) static analysis tool, and executes with the local Python interpreter.
-If there are no errors, callysto is used to generate the source script into an `.ipynb`, which is copied into the Terra
-workspace bucket.
+## Testing Notebooks
+Tests are performed in Docker containers of the expected notebook deployment platform, e.g. Terra, with the command
+```
+make test
+```
 
-Publishing requires Google client authorization, as well as Terra access to the target workspace. Google authorization
-is obtained with the command
+Tests can also be performed on individual notebooks
 ```
-gcloud auth login
+make test/byod
 ```
 
+These recipes pass the source script through the [flake8](https://flake8.pycqa.org/en/latest/) linter and
+[mypy](https://mypy.readthedocs.io/en/stable/) static analysis tool, and executes with a Docker container that is
+typical of Terra notebook runtime environments for Python. If there are no errors, callysto is used to generate the
```

Comment:
Suggestion: `callysto` might benefit as a text link to the repo.

Reply:
will add